conjure/conjure.go

```diff
@@ -69,10 +69,14 @@ func GenerateOutputFiles(conjureDefinition spec.ConjureDefinition, cfg OutputCon
 		}
 		if len(pkg.Unions) > 0 {
 			unionFile := newJenFile(pkg, def)
+			go1Dot18UnionFile := newJenFile(pkg, def)
+			go1Dot18UnionFile.Comment("//go:build go1.18")
 			for _, union := range pkg.Unions {
 				writeUnionType(unionFile.Group, union, cfg.GenerateFuncsVisitor)
+				writeUnionTypeWithGenerics(go1Dot18UnionFile.Group, union)
 			}
 			files = append(files, newGoFile(filepath.Join(pkg.OutputDir, "unions.conjure.go"), unionFile))
+			files = append(files, newGoFile(filepath.Join(pkg.OutputDir, "unions.conjure_go1.18.go"), go1Dot18UnionFile))
```

Comment:
Can we change this to something like `unions_generics.conjure.go` or `unions_go118.conjure.go`?

For consistency, I'd like to maintain the `.conjure.go` suffix for all Conjure-generated files, and would also like to avoid using `go1.18` as part of the name since I'd like to reserve the dot (`.`) character for semantic delineation of path parts.

Reply:
Makes sense, done - I chose `unions_generics.conjure.go`